contribs/gnofaucet/gh_api.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/google/go-github/v74/github"
+)
+
+type GithubClient interface {
+	ListRepositoryEvents(ctx context.Context, owner, repo string, page int) ([]*github.Event, *github.Response, error)
+	ListPullRequests(ctx context.Context, owner string, repo string, cursor string) ([]PullRequest, string, error)
+	ListIssues(ctx context.Context, owner string, repo string, page int) ([]*github.Issue, *github.Response, error)
+}
+
+var _ GithubClient = &GithubClientImpl{}
+
+type GithubClientImpl struct {
+	cl  *github.Client
+	gql graphql.Client
+}
+
+// ListIssues implements GithubClient.
+func (lre *GithubClientImpl) ListIssues(ctx context.Context, owner string, repo string, page int) ([]*github.Issue, *github.Response, error) {
+	return lre.cl.Issues.ListByRepo(ctx, owner, repo, &github.IssueListByRepoOptions{
+		Sort:      "created",
+		Direction: "asc",
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+			Page:    page,
+		},
+	})
+}
+
+// ListPullRequests implements GithubClient.
+func (lre *GithubClientImpl) ListPullRequests(ctx context.Context, owner string, repo string, cursor string) ([]PullRequest, string, error) {
+	data, err := getPullRequests(ctx, lre.gql, owner, repo, cursor)
```

Comment:
Naive question: why use GraphQL instead of just listing PRs with go-github like you did for the issues above? What's the benefit?

Reply:
They have different functionalities implemented on each one. For example, there is no way to fetch events using the GraphQL API, and there is no way to fetch all reviews from a PR on a single request using the standard REST API.